pkg/v4/resource/chart/resource.go

```diff
@@ -165,3 +165,13 @@ func toResourceState(v interface{}) (ResourceState, error) {
 
 	return *resourceState, nil
 }
+
+func isChartModified(a, b ResourceState) bool {
+	if a.ReleaseVersion != b.ReleaseVersion {
```

Comment:
`ResourceState` is a struct and as soon as somebody extends it with a function, slice or map the equality check will no longer work. It would be cool to implement an `Equals` method or do the specific testing here or just use `reflect.DeepEqual`.

Reply:
Here we are comparing string fields, correct me if I'm wrong but, no matter how `ResourceState` changes, we check the `ReleaseVersion` field and it wouldn't change if we modify the struct in other ways. Is that correct, am I missing something?

Reply:
You are right. Vision and thought didn't match here apparently. :D